spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -2701,4 +2701,20 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       }
     }
   }
+
+  test("array_compact") {
+    assume(isSpark34Plus)
+    Seq(true, false).foreach { dictionaryEnabled =>
+      withTempDir { dir =>
+        val path = new Path(dir.toURI.toString, "test.parquet")
+        makeParquetFileAllTypes(path, dictionaryEnabled = dictionaryEnabled, n = 10000)
+        spark.read.parquet(path.toString).createOrReplaceTempView("t1")
+
+        checkSparkAnswerAndOperator(
+          sql("SELECT array_compact(array(_2)) FROM t1 WHERE _2 IS NULL"))
+        checkSparkAnswerAndOperator(
+          sql("SELECT array_compact(array(_2)) FROM t1 WHERE _2 IS NOT NULL"))
```

Comment:
I'd like to see tests covering the case where the array contains both null and non null values, but we can address that as part of https://github.com/apache/datafusion-comet/issues/1269.

Reply:
One of the disadvantages of `makeParquetFileAllTypes` is that if `_2` is null, then all other columns are also null.

Reply:
+1 for more coverage